src/nvim/api/buffer.c

```diff
@@ -636,6 +634,297 @@ ArrayOf(Integer, 2) buffer_get_mark(Buffer buffer, String name, Error *err)
   return rv;
 }
 
+/// Returns mark info at the given position or mark index
+///
+/// @param buffer The buffer handle
+/// @param namespace a identifier returned previously with namespace_create
+/// @param id [row, col] or mark_id of mark
+/// @param[out] err Details of an error that may have occurred
+/// @return [mark_id, row, col]
+ArrayOf(Object) buffer_mark_index(Buffer buffer, Integer namespace, Object id, Error *err)
```

Comment:
@timeyyy I personally prefer that convention for the same reason you mention, but I'm not sure if anyone else does. If others are ok with that, then we should do it before  #4934 so that we can rename all API methods follow that convention:

    nvim_{namespace}_{noun}_{verb}_{arbitrary qualifiers}